### PR TITLE
Handle silent failures in membership materialization cascade

### PR DESCRIPTION
**Pull request by delano**
## Summary

Improve the robustness of the `MaterializePlans` operation to handle cases where membership materialization fails silently (returns `false` rather than raising an exception). This can occur when a membership's role has no corresponding entry in the `ROLE_ENTITLEMENTS` catalog—e.g., a role that was removed from the system.

### Changes

**Operation (`materialize_plans.rb`)**
- Modified `run_cascade` to check the return value of `materialize_for_role!`, not just catch exceptions
- When `materialize_for_role!` returns `false`, the membership is now counted as a failure rather than silently succeeding
- Added detailed error logging and cascade detail tracking for false-return cases
- Added injectable `membership_loader` parameter to allow tests to inject membership lists without stubbing internal batch primitives

**Tests**
- Added comprehensive integration spec (`materialize_plans_integration_spec.rb`) covering the `membership_loader` seam with real Redis and Familia models
- Extended unit tests in `materialize_plans_spec.rb` to verify false-return handling and the new `membership_loader` parameter
- Updated CLI integration spec with additional test cases:
  - Dry-run mode (no writes)
  - Plan filtering
  - Orgs with no plan
  - Orgs on missing plans
  - Cascades with zero active memberships
  - Mixed pending and active memberships
  - Bad-role memberships (the real-world scenario for false returns)
- Fixed test isolation issues by explicitly requiring `controllers/base.rb` after `billing_spec_helper`

### Cascade Invariant

The operation now maintains the invariant: **never mask partial success**. If any membership in a cascade fails (whether by exception or silent no-op), the entire org is counted as FAILED, with partial counts captured in the error reason for audit and retry decisions.

## Test Plan

All new and modified tests pass:
- `materialize_plans_integration_spec.rb` — 2 new integration tests covering the `membership_loader` seam
- `materialize_plans_spec.rb` — 3 new unit tests for false-return handling and loader injection
- `plans_materialize_command_integration_spec.rb` — 6 new CLI integration tests covering edge cases and cascade scenarios

Existing tests continue to pass, confirming backward compatibility.

https://claude.ai/code/session_01Y9njeX34obiG6vVgFoxGuE

### CODE REVIEW

File: apps/web/billing/operations/materialize_plans.rb [hunk 1]
```diff
@@ -219,17 +231,37 @@ def record_org_write_failure(org, ex)
...
       # the --verbose renderer and for richer audit logs. The webhook path
       # continues to use the org's aggregate method.
+      #
+      # The membership list comes from @membership_loader (defaults to
+      # OrganizationMembership.active_for_org) so cascade specs can inject
+      # memberships through a stable seam instead of stubbing active_for_org's
+      # internal batch primitive.
       def run_cascade(org)
-        memberships = Onetime::OrganizationMembership.active_for_org(org)
+        memberships = @membership_loader.call(org)
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/web/billing/operations/materialize_plans.rb]: Addressed in 0ed39ddb — the loader result is now guarded with `(@membership_loader.call(org) || []).compact`, so a nil return is treated as an empty cascade.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/billing/operations/materialize_plans.rb [hunk 2]
```diff
@@ -219,17 +231,37 @@ def record_org_write_failure(org, ex)
...
         details     = []
         succeeded   = 0
         failed      = 0
 
         memberships.each do |membership|
           begin
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/web/billing/operations/materialize_plans.rb:251]: Addressed in 0ed39ddb. I used `.compact` on the loader result rather than a per-iteration `next if membership.nil?`: `(@membership_loader.call(org) || []).compact`. That drops nil entries before iteration (so neither `materialize_for_role!` nor the rescue block's `membership.objid`/`.role` can hit nil) and also keeps the cascade `total` count accurate. Added doubles specs for both a nil return and nil entries.

---
_Generated by [Claude Code](https://claude.ai/code)_
